lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -102,6 +101,17 @@ def set_token(self, token: str) -> None:
         """Set the Vault token for authentication."""
         self._client.token = token
 
+    def approle_login(self, role_id: str, secret_id: str) -> None:
+        """Authenticate to Vault using AppRole."""
+        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+
+    def configure_charm_access_policy(self, name: str) -> None:
+        """Create/update a policy within vault for the charm to access vault."""
+        with open("src/templates/charm_policy.hcl", "r") as f:
```

Comment:
Seems odd to me that we enforce the location here, but I tried to keep it consistent with https://github.com/canonical/vault-k8s-operator/blob/70431d8c412912270f39ee84e00bc32c67814736/lib/charms/vault_k8s/v0/vault_client.py#L250 for now.